packages/edit-site/src/components/layout/index.js

```diff
@@ -296,82 +281,82 @@ export default function Layout() {
 
 					<SavePanel />
 
-					{ showCanvas && (
-						<>
-							{ isListPage && <PageMain /> }
-							{ isEditorPage && (
-								<div className="edit-site-layout__canvas-container">
-									{ canvasResizer }
-									{ !! canvasSize.width && (
-										<motion.div
-											whileHover={
-												isEditorPage &&
-												canvasMode === 'view'
-													? {
-															scale: 1.005,
-															transition: {
-																duration:
-																	disableMotion
-																		? 0
-																		: 0.5,
-																ease: 'easeOut',
-															},
-													  }
-													: {}
+					{ areas.content && canvasMode !== 'edit' && (
```

Comment:
This is the new "content" area.

Reply:
For someone with limited exposure to site editor code, like myself, the fact that the logic to decide what to render was split in so many places made it hard to navigate and understand. I welcome an approach that centralizes the routing logic and maps it to render areas.

With that in mind, why the `canvasMode` shouldn't be absorbed by the `useLayout` hook?

Reply:
> With that in mind, why the canvasMode shouldn't be absorbed by the useLayout hook?

Can you clarify a bit more what you have in mind?

Reply:
It'd be great to see something along these lines:

```js
// The useLayoutAreas hook centralizes all logic about which area is avaliable.
function useLayoutAreas() {
    if ( path === '/page' && canvasMode === 'edit' ) {
        return { areas: { /* ... */ } };
    }

    if ( path === '/page' ) {
        return { areas: { /* ... */ } };
    }

    // etc.
}

// The Layout component composes the layout.
export default function Layout() {
    const { areas } = useLayoutAreas();

    return (
        <div className="edit-site-layout">
            { areas.sidebar && <div className="edit-site-layout__sidebar">{ areas.sidebar }</div> }
            { areas.content && <div className="edit-site-layout__content">{ areas.sidebar }</div> }
            { areas.preview && <div className="edit-site-layout__preview">{ areas.sidebar }</div> }
        </div>
    );
}
```

Reply:
I agree that's the goal. 

 - We can't do sidebar now for different reasons: the navigation component need to be controlled (which is not at the moment).
 - We can't ditch the canvas mode entirely (maybe later?) because it's a state that is used in very different places.